Made a docker-compose file to easily run an archive node (#4060)
CodaProtocol/coda: frontend/website/docs/developers/archive-node.md

## Background

> Since Coda nodes are by default succinct, if you need to preserve historical state, you'll want to run an `archive node`. The easiest way to run the `archive node` is by the prod archive node docker-compose file.

Reviewer: ```suggestion
Since Coda nodes are by default succinct, if you need to preserve historical state, you'll want to run an `archive node`. The easiest way to run the `archive node` is by using the archive node docker-compose file:
```

> 

Reviewer: ```suggestion
```

> Here are required steps to achieve this:

Reviewer: ```suggestion
```

> 1. Clone the coda repository:

Reviewer: ```suggestion
1. Clone the Coda repository:
```

    `git clone https://github.com/CodaProtocol/coda.git`
2. Install `docker-compose`. You can find installation instructions for `docker-compose` on [the docker website] (https://docs.docker.com/compose/install/).

> 3. Ensure that ports 3085, 3086, 9000 and 5432 are open. These ports correspond to ports for the daemon, archive, Hasura and Postgres servers to run.

Reviewer: ```suggestion
3. Ensure that ports 3085, 3086, 9000, and 5432 are open. These ports correspond to ports where the Coda daemon, archive process, Hasura, and Postgres servers will run.
```

Reviewer: @wu-s-john maybe it may be useful to show how to forward ports here? Or at least add a link to instructions

> Start the daemon using this command to run an archive process:

Reviewer: ```suggestion
Start the Coda daemon using this command to run an archive process:
```

> 

Reviewer: ```suggestion
```

```bash
docker-compose -f scripts/archive/docker-compose.prod.yml
```

### Note

> Running the archive node may not be compatible with the current testnet as the deployment configurations of the archive node is on a unstable develop branch.

Reviewer: ```suggestion
Running the archive node may not be compatible with the current testnet as the deployment configurations of the archive node is the develop branch, which may have features not supported currently by the testnet.
```

## Architecture

<img width="100%" src="https://cdn.codaprotocol.com/website/static/img/archive-node-diagram-c5424a1f7a6993488f3337f81b7b929ae2cfeb9d2656413106606f7ecc804e06.png" alt="Coda archive node architecture diagram" />

> **Writes**

Reviewer: 👍 thank you!
